### PR TITLE
Add basic player

### PR DESCRIPTION
**Pull request by minicarter**
# Add basic player
* Add Player anim idle + move
* Add player sprites for anims
* Add player prefab
* Add player tutorial scene
* Fix camera height
* Move scripts in Scripts/...

### CODE REVIEW

File: Assets/Scripts/Common/Singleton.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using UnityEngine;
+
+public class Singleton<T> : MonoBehaviour where T : Component
+{
+    private static T _instance;
+
+    public static bool m_isQuitting;
```
Comment (tris790) [inline Assets/Scripts/Common/Singleton.cs]: enleve ca

File: Assets/Scripts/Components/MovementComponent.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    // Start is called before the first frame update
+    void Start()
+    {
+        _rigidbody = GetComponent<Rigidbody2D>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        ProcessInputs();
```
Comment (tris790) [inline Assets/Scripts/Components/MovementComponent.cs:26]: A valider que on veut process les inputs dans Update ou FixedUpdate

File: Assets/Scripts/Components/MovementComponent.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    public void Move()
+    {
+        _rigidbody.velocity = new Vector2(_moveDirection.x * moveSpeed, _moveDirection.y * moveSpeed);
+    }
+
+    public void Rotate()
+    {
+        Vector3 mousePos = Input.mousePosition;
+        mousePos.z = 5.23f;
```
Comment (tris790) [inline Assets/Scripts/Components/MovementComponent.cs]: ?

File: Assets/Scripts/Map/MapGenerator.cs [hunk 4]
```diff
@@ -1,139 +1,139 @@
...
+{
+    public enum DrawMode { NoiseMap,ColorMap };
+    public DrawMode drawMode;
+    public Noise.NormalizeMode normalizeMode;
+
+
+    //Propriété de la map pouvant etre éditer depuis l'editeur Unity
+    public int seed;
+    //must be a multiple de 10
+    public const int mapChunkSize=40;
```
Comment (tris790) [inline Assets/Scripts/Map/MapGenerator.cs]: ctrl+k+d

File: Assets/Scripts/Map/MapGenerator.cs [hunk 5]
```diff
@@ -1,139 +1,139 @@
...
+            display.DrawTexture(TextureGenerator.TextureFromColorMap(mapData.colorMap, mapWidth, mapHeight));
+        }
+    }
+
+    struct MapThreadInfo<T,DrawMode>
+    {
+        public readonly Action<T,DrawMode> callback;
+        public readonly T param;
+        public readonly DrawMode param2;
+        public MapThreadInfo(Action<T,DrawMode> callback,T param,DrawMode param2)
```
Comment (tris790) [inline Assets/Scripts/Map/MapGenerator.cs]: ctrl+k+d

File: Assets/Scripts/Map/Noise.cs [hunk 6]
```diff
@@ -1,88 +1,88 @@
...
-    }
-}
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public static class Noise
+{
+    public enum NormalizeMode { Local, Global }
+    public static float[,] GenerateNoiseMap(int width, int height,int seed, float scale,int octaves, float persistance, float lacunarity,Vector2 offset, NormalizeMode normalizeMode)
```
Comment (tris790) [inline Assets/Scripts/Map/Noise.cs]: ctrl+k+d

File: Assets/Scripts/Components/MovementComponent.cs (hunk 2 above)
Comment (MYSTACK555) [reply Assets/Scripts/Components/MovementComponent.cs:26]: apres lecture de https://docs.unity3d.com/ScriptReference/MonoBehaviour.FixedUpdate.html moi je pense que c'est good comme ca

**Comment by tris790** [comment]
🥓

**Pull request merged**